app/components/work_packages/progress/work_based/modal_body_component.html.erb

```diff
@@ -7,7 +7,7 @@
         data: { "application-target": "dynamic",
                 controller: "work-packages--progress--focus-field" }) do |f| %>
     <%= flex_layout do |modal_body| %>
-      <% modal_body.with_row do |_fields| %>
+      <% modal_body.with_row(classes: "horizontal-form-group") do |_fields| %>
         <%= render(WorkPackages::ProgressForm.new(f, work_package:, mode:, focused_field:)) %>
```

Comment:
It feels weird, that this class is on the body as it actually belongs to the form. I know, that the Primer:: Group doesn't apply the system_arguments but maybe there is still an alternative way. Maybe you could add a parameter `mobile_alignment` (or whatever) and inside the component render a div around the form which is only responsible to display the class if the parameter is set ?